### PR TITLE
SISRP-29182 Remove obsolete APIs before they cause more breakage

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.berkeley.edu/browse/SISRP-29182

"18 additions and 6,431 deletions" -- now, _that_ feels like a good day's work!

### CODE REVIEW

**Comment by raydavis** [comment]
Amended to fix jscs fail.

File: app/models/finaid/fin_aid_year.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 module Finaid
+  # TODO : DROP THIS TABLE ONE RELEASE AFTER REMOVAL OF LEGACY FINAID SUPPORT.
```
Comment (lyttam) [inline app/models/finaid/fin_aid_year.rb]: @raydavis do we have a corresponding Jira to remember this by, or some other way of tracking/scheduling this work?

Comment (raydavis) [inline app/models/finaid/fin_aid_year.rb]: Yes, @lyttam , but thank you for asking: https://jira.berkeley.edu/browse/SISRP-29183

**Comment by raydavis** [comment]
Unshockingly, @lyttam 's massive PR conflicts with my massive PR. I'll amend ASAP

**Comment by raydavis** [comment]
Amended.

**Comment by lyttam** [comment]
Wow, @raydavis, this is a lot of cleanup!  So, just so I understand, regstatus and myfinaid were giving us pre-Fall16 "current and future" terms - and, now that we're past Fall 2016, there would be no more terms for those APIs to give?

**Comment by raydavis** [comment]
Right, @lyttam . None of those APIs were reliable for past history (although they failed in different and sometimes misleading ways).

File: src/assets/templates/myfinances.html [hunk 2]
```diff
@@ -12,7 +12,6 @@ <h1 class="cc-heading-page-title">My Finances</h1>
 
     <div class="medium-6 large-4 cc-column-2 columns">
       <div data-ng-include="'widgets/finaid_summary.html'" data-ng-if="api.user.profile.features.csFinAid && !academicStatus.roles.summerVisitor"></div>
-      <div data-ng-include="'widgets/finaid_activity.html'" data-ng-if="!api.user.profile.isApplicantOnly"></div>
```
Comment (nkutub) [inline src/assets/templates/myfinances.html]: @raydavis, so reading this I'm assuming that there is no alternative source for finaid activity and that it is just not a feature we are providing anymore?

Comment (raydavis) [inline src/assets/templates/myfinances.html]: @nkutub , the old FinAid API & UX has been replaced by new CS APIs & UX. I wasn't much involved in the transition so I can't say item-by-item what functionality might have been added or dropped. Either way, even in the pre-CS world, the finaid_activity card for the 2015-2016 academic year would be gone by now. Does that clarify?

Comment (nkutub) [inline src/assets/templates/myfinances.html]: @raydavis good enough for now, I will eventually get my arms around all the functions CC provides and its data sources

Comment (raydavis) [inline src/assets/templates/myfinances.html]: Be careful not to tear a ligament!

**Comment by davidlimcheng** [comment]
LGTM @lyttam @nkutub.  Good stuff @raydavis !

**Pull request merged**